### PR TITLE
 [ADD] Option state buttons and messages

### PR DESCRIPTION
**Pull request by eilst**

### CODE REVIEW

File: project_event/models/project_task.py [hunk 1]
```diff
@@ -239,3 +241,15 @@ def action_read(self):
     def action_draft(self):
         self.write({'task_state': 'draft'})
         self.draft_resources_reservation()
+
+    def send_message(self, action):
+        switcher = {
+            "draft": "",
+            "option": "The following are Optional and no longer on your calendars",
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: @eilst we will allow double reservation?

Comment (rimbendhaou) [reply project_event/models/project_task.py]: @eilst The following are or is?

File: project_event/models/project_task.py [hunk 2]
```diff
@@ -239,3 +241,15 @@ def action_read(self):
...
+            "requested": "The following is requested",
+            "accepted": "",
+            "read": "",
+            "done": "",
+            "canceled": "The following is canceled and no longer on your calendars"
+        }
+        switcher.get(action)
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: @eilst when it will send the message?

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_event/models/project_task.py [hunk 3]
```diff
@@ -162,6 +162,8 @@ def action_request(self):
     def action_option(self):
         if self.activity_task_type == 'task':
             self.draft_resources_reservation()
+        if self.task_state in ['requested', 'read', 'accepted']:
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: @eilst I think it should be under the if self.activity_task_type == 'task' statement

File: project_event/models/project_task.py [hunk 4]
```diff
@@ -239,3 +241,32 @@ def action_read(self):
     def action_draft(self):
         self.write({'task_state': 'draft'})
         self.draft_resources_reservation()
+
+    def get_message_body(self, action):
+        switcher = {
+            "draft": "",
+            "option": "The following are Optional and no longer on your calendars",
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: @eilst too much long line (PEP8). Add the _('') for translation

File: project_event/models/project_task.py [hunk 5]
```diff
@@ -239,3 +241,32 @@ def action_read(self):
...
+            "read": "",
+            "done": "",
+            "canceled": "The following is canceled and no longer on your calendars"
+        }
+        return switcher.get(action)
+
+    def get_message(self, action):
+        return {
+            'body': self.get_message_body(action),
+            'channel_ids': [(6, 0, [5])],
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: @eilst put the external id of the channel not the id

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_event/models/project_task.py (hunk 3 above)
Comment (eilst) [reply project_event/models/project_task.py]: right

Comment (eilst) [reply project_event/models/project_task.py]: right

**Review by rimbendhaou — APPROVED** [review]

**Pull request merged**